File: TelerikUIExamples/TelerikUIExamples/SideDrawerPositions.m
//
//  SideDrawerPositions.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import "SideDrawerPositions.h"
#import "SideDrawerHeaderView.h"

@implementation SideDrawerPositions
{
    UIScrollView *_scrollView;
    CGFloat _buttonY;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _scrollView = [[UIScrollView alloc] initWithFrame:CGRectZero];
    _scrollView.scrollEnabled = YES;
    
    [self.sideDrawerView.mainView addSubview:_scrollView];
    
    [self addButtonWithTitle:@"Left" target:self selector:@selector(leftSideDrawer)];
    [self addButtonWithTitle:@"Right" target:self selector:@selector(rightSideDrawer)];
    [self addButtonWithTitle:@"Top" target:self selector:@selector(topSideDrawer)];
    [self addButtonWithTitle:@"Bottom" target:self selector:@selector(bottomSideDrawer)];
    
    self.sideDrawer.fill = [TKSolidFill solidFillWithColor:[UIColor grayColor]];
    self.sideDrawer.transition = TKSideDrawerTransitionTypeReveal;
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    _scrollView.frame = CGRectMake(0, 44, CGRectGetWidth(self.view.bounds), CGRectGetHeight(self.view.bounds) - 44);
}

- (void)leftSideDrawer
{
    TKSideDrawer *sideDrawer = self.sideDrawer;
    sideDrawer.position = TKSideDrawerPositionLeft;
    sideDrawer.headerView = [[SideDrawerHeaderView alloc] initWithButton:NO target:nil selector:nil];
    [self showSideDrawer];
}

- (void)rightSideDrawer
{
    TKSideDrawer *sideDrawer = self.sideDrawer;
    sideDrawer.position = TKSideDrawerPositionRight;
    sideDrawer.headerView = [[SideDrawerHeaderView alloc] initWithButton:YES target:self selector:@selector(dismissSideDrawer)];
    [self showSideDrawer];
}

- (void)topSideDrawer
{
    TKSideDrawer *sideDrawer = self.sideDrawer;
    sideDrawer.position = TKSideDrawerPositionTop;
    sideDrawer.headerView = [[SideDrawerHeaderView alloc] initWithButton:NO target:nil selector:nil];
    [((TKSideDrawerTableView *)sideDrawer.content) setContentOffset:CGPointZero animated:NO];
    sideDrawer.allowScroll = YES;
    [self showSideDrawer];
}

- (void)bottomSideDrawer
{
    TKSideDrawer *sideDrawer = self.sideDrawer;
    sideDrawer.position = TKSideDrawerPositionBottom;
    sideDrawer.headerView = [[SideDrawerHeaderView alloc] initWithButton:YES target:self selector:@selector(dismissSideDrawer)];
    [((TKSideDrawerTableView *)sideDrawer.content) setContentOffset:CGPointZero animated:NO];
    sideDrawer.allowScroll = YES;
    [self showSideDrawer];
}

- (void)addButtonWithTitle:(NSString *)title target:(id)target selector:(SEL)selector
{
    CGSize titleSize = [title sizeWithAttributes:@{ NSFontAttributeName : [UIFont systemFontOfSize:18] }];
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(15, 15 + _buttonY, titleSize.width * 2, 44)];
    button.titleLabel.font = [UIFont systemFontOfSize:14];
    [button.layer setBorderWidth:1.0];
    [button.layer setBorderColor:[UIColor whiteColor].CGColor];
    [button.layer setCornerRadius:3.f];
    [button setTitle:title forState:UIControlStateNormal];
    [button setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [button addTarget:target action:selector forControlEvents:UIControlEventTouchUpInside];
    [_scrollView addSubview:button];
    _buttonY += 50;
    _scrollView.contentSize = CGSizeMake(fmax(CGRectGetWidth(button.frame), _scrollView.contentSize.width), _buttonY + 15);
}

#pragma mark TKSideDrawerDelegate

- (void)sideDrawer:(TKSideDrawer *)sideDrawer updateVisualsForSection:(NSInteger)sectionIndex
{
    TKSideDrawerSection *section = sideDrawer.sections[sectionIndex];
    section.style.contentInsets = UIEdgeInsetsMake(0, -15, 0, 0);
}

- (void)sideDrawer:(TKSideDrawer *)sideDrawer updateVisualsForItem:(NSInteger)item inSection:(NSInteger)section
{
    TKSideDrawerItem *currentItem = ((TKSideDrawerSection *)sideDrawer.sections[section]).items[item];
    currentItem.style.contentInsets = UIEdgeInsetsMake(0, -10, 0, 0);
    currentItem.style.separatorColor = [TKSolidFill solidFillWithColor:[UIColor clearColor]];
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SideDrawerPositions.swift
//
//  SideDrawerPositions.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerPositions: SideDrawerGettingStarted {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.createButton("Left", target: self, selector: #selector(leftSideDrawer), origin: CGPointMake(15, 80))
        self.createButton("Right", target: self, selector: #selector(rightSideDrawer), origin: CGPointMake(15, 130))
        self.createButton("Top", target: self, selector: #selector(topSideDrawer), origin: CGPointMake(15, 180))
        self.createButton("Bottom", target: self, selector: #selector(bottomSideDrawer), origin: CGPointMake(15, 230))
        
        let sideDrawer = self.sideDrawerView.sideDrawers[0]
        sideDrawer.transition = TKSideDrawerTransitionType.Reveal
        
        // >> drawer-blur-swift
        sideDrawer.fill = TKSolidFill(color: UIColor.grayColor())
        sideDrawer.style.blurType = TKSideDrawerBlurType.None
        // << drawer-blur-swift
    }
    
    func createButton(title: String, target: AnyObject, selector: Selector, origin: CGPoint) {
        let titleString: NSString = title as NSString
        let titleSize = titleString.sizeWithAttributes([NSFontAttributeName : UIFont.systemFontOfSize(18.0)])
        let button = UIButton(frame: CGRectMake(origin.x, origin.y, titleSize.width * 1.5, 44))
        button.titleLabel?.font = UIFont.systemFontOfSize(14)
        button.layer.borderWidth = 1.0
        button.layer.borderColor = UIColor.whiteColor().CGColor
        button.layer.cornerRadius = 3.0
        button.setTitle(title, forState: UIControlState.Normal)
        button.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        button.addTarget(self, action: selector, forControlEvents: UIControlEvents.TouchUpInside)
        self.sideDrawerView.mainView.addSubview(button)
    }
    
    func leftSideDrawer() {
        let sideDrawer = self.sideDrawerView.sideDrawers[0]
        // >> drawer-position-swift
        sideDrawer.position = TKSideDrawerPosition.Left
        // << drawer-position-swift
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        self.showSideDrawer()
    }
    
    func rightSideDrawer() {
        let sideDrawer = self.sideDrawerView.sideDrawers[0]
        sideDrawer.position = TKSideDrawerPosition.Right
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: #selector(dismissSideDrawer))
        self.showSideDrawer()
    }
    
    func topSideDrawer() {
        let sideDrawer = self.sideDrawerView.sideDrawers[0]
        sideDrawer.position = TKSideDrawerPosition.Top
        (sideDrawer.content as! TKSideDrawerTableView) .setContentOffset(CGPointZero, animated: false)
        sideDrawer.headerView = SideDrawerHeaderView(addButton: false, target: nil, selector: nil)
        sideDrawer.allowScroll = true
        self.showSideDrawer()
    }
    
    func bottomSideDrawer() {
        let sideDrawer = self.sideDrawerView.sideDrawers[0]
        sideDrawer.position = TKSideDrawerPosition.Bottom
        (sideDrawer.content as! TKSideDrawerTableView).setContentOffset(CGPointZero, animated: false)
        sideDrawer.allowScroll = true
        sideDrawer.headerView = SideDrawerHeaderView(addButton: true, target: self, selector: #selector(dismissSideDrawer))
        self.showSideDrawer()
    }
}
